design-proposals/virtiofs-support/virtiofs-feature-gates.md

```diff
@@ -0,0 +1,69 @@
+# Overview
+Virtiofs allows sharing Secrets, ConfigMaps, ServiceAccounts and DownwardAPI (a.k.a., ConfigVolumes) in a dynamic way,
+i.e., any change in those objects are reflected in the VirtualMachine (VM) without restarting the VM. Moreover, PVCs may
+also be shared using virtiofs allowing multiple readers and writers in the shared PVCs.
+
+## Motivation
+Currently, the usage of feature gates are mixed with the feature control. The feature gate `ExperimentalVirtiofsSupport`
+controls whether virtiofsd runs as rootful container inside the launcher pod or not. Moreover, there is no way to
+disable it completely and this feature is not in GA.
+
+In order to GA virtiofs, VMs should be able to live migrate while sharing data with virtiofs. The virtiofsd’s live
+migration support will distinguish between different use cases:  sharing ConfigMaps, Secrets, DownwardAPI and
+ServiceAccounts and sharing PVCs. 
+
+This distinction will be useful because in order to fully support sharing ConfigVolumes and PVCs while allowing
```

Comment:
The first sentence actually doesn't justify the distinction. I would remove the "the distinction will` and move to the part where you explain that different set of changes are required for ConfigVolumes GA and PVC GA...

Reply:
Adjusted.